subgraphs/rocket-pool/src/updaters/usageMetrics.ts

```diff
@@ -0,0 +1,111 @@
+import { Address, ethereum } from "@graphprotocol/graph-ts";
+import { getOrCreateProtocol } from "../entities/protocol";
+import {
+  Account,
+  ActiveAccount,
+  UsageMetricsDailySnapshot,
+  UsageMetricsHourlySnapshot,
+} from "../../generated/schema";
+import { SECONDS_PER_DAY, SECONDS_PER_HOUR } from "../utils/constants";
+
+export function updateUsageMetrics(block: ethereum.Block, from: Address): void {
+  const accountId = from.toHexString();
+  let account = Account.load(accountId);
+  if (!account) {
+    account = new Account(accountId);
+    account.save();
+
+    const protocol = getOrCreateProtocol();
+    protocol.cumulativeUniqueUsers += 1;
+    protocol.save();
+  }
+
+  const protocol = getOrCreateProtocol();
+  const usageMetricsDailySnapshot = getOrCreateUsageMetricsDailySnapshot(block);
+  const usageMetricsHourlySnapshot =
+    getOrCreateUsageMetricsHourlySnapshot(block);
+
+  usageMetricsDailySnapshot.dailyTransactionCount += 1;
+  usageMetricsHourlySnapshot.hourlyTransactionCount += 1;
+
+  usageMetricsDailySnapshot.cumulativeUniqueUsers =
+    protocol.cumulativeUniqueUsers;
+  usageMetricsHourlySnapshot.cumulativeUniqueUsers =
+    protocol.cumulativeUniqueUsers;
+
+  const dailyActiveAccountId = (block.timestamp.toI64() / SECONDS_PER_DAY)
+    .toString()
+    .concat("-")
+    .concat(accountId);
+
+  let dailyActiveAccount = ActiveAccount.load(dailyActiveAccountId);
+  if (!dailyActiveAccount) {
+    dailyActiveAccount = new ActiveAccount(dailyActiveAccountId);
+    dailyActiveAccount.save();
+
+    usageMetricsDailySnapshot.dailyActiveUsers += 1;
+    usageMetricsHourlySnapshot.hourlyActiveUsers += 1;
```

Comment:
you need a separate check for hourlyActiveUsers

Reply:
WDYM? Each time update is called it updates the hourly and daily transaction account and then saves it in a diff object but those are maintained separately. The separate mechanism handles the hours and days elapsing at separate intervals.

Reply:
Well you are counting the `dailyActiveUsers` and incrementing the `hourlyAcitveUsers`. You need a separate check for an `hourlyActiveAccount` I would think.

The way it is right now I belv it would count a user who used the protocol for the first time in the daily / hourly (that is good), but then if that same user made another transaction in the next hour (same day) it would not count that since that `dailyActiveAccount` has already been created.